src/components/dashboard/components/race/raceContainer.jsx

```diff
@@ -13,18 +13,21 @@ class RaceContainer extends Component {
     this._loadCampaignData = this._loadCampaignData.bind(this);
   }
 
-  _loadCampaignData(candidateId, campaignId) {
-    Client.getCandidate({
+  async _loadCampaignData(candidateId, campaignId) {
+
+    const json = await Client.getCandidate({
       id: candidateId,
       campaigns: [{ campaignId: campaignId }]
-    }).then(data => {
-      let scores = this.state.scores;
-      scores[campaignId] = data.campaigns[0];
-      this.setState({
-        scores: scores,
-        selectedCampaign: campaignId
-      });
-    });
+    })
+
+    let scores = this.state.scores;
```

Comment:
since line 18 through 22 are waiting, could the lines 23 through 30 execute before the await complete? see if you find any doc around async/await and the react lifecycle